ownCloud/Client/Actions/Actions+Extensions/UploadPhotosAction.swift

```diff
@@ -75,16 +79,52 @@ class UploadPhotosAction: UploadBaseAction {
 		}
 	}
 
-	func presentImageGalleryPicker() {
+	private func presentImageGalleryPicker() {
 		OnMainThread {
 			if let viewController = self.context.viewController {
 				let photoAlbumViewController = PhotoAlbumTableViewController()
 				photoAlbumViewController.selectionCallback = { (assets) in
-					for asset in assets {
-						self.upload(asset: asset)
-					}
 
 					self.completed()
+
+					let queue = DispatchQueue.global(qos: .userInitiated)
+
+					queue.async {
+
+						self.core?.perform(inRunningCore: { (runningCoreCompletion) in
+
+							let uploadGroup = DispatchGroup()
+							var uploadFailed = false
+
+							for asset in assets {
+								if uploadFailed == false {
+									// Upload image on a background queue
+									uploadGroup.enter()
```

Comment:
(1/3) The group is always entered…

Reply:
@felix-schwarz made sure that enter and leave are balanced